WKWebView.ios.js

```diff
@@ -308,12 +308,11 @@ class WKWebView extends React.Component {
       WKWebViewManager.startLoadWithResult(!!shouldStart, event.nativeEvent.lockIdentifier);
     });
 
-    let source = {};
-    if (this.props.source && typeof this.props.source == 'object') {
-      source = Object.assign({}, this.props.source, {
-        sendCookies: this.props.sendCookies,
-        customUserAgent: this.props.customUserAgent || this.props.userAgent
-      });
+    let source = this.props.source || {};
+    if (typeof source == 'object') {
+      source.sendCookies = this.props.sendCookies;
```

Comment:
`source` is still referencing to `this.props.source`

Reply:
Could be fix by setting source like this I guess:
```js
let source = { ...this.props.source };
```

Reply:
Won't it break scenario with `source={require('...')}`?

Reply:
Indeed.
That's probably the if condition that fails, so how about a check like this instead:
```js
if (typeof source == 'object' && source.constructor == Object) { //...
```

Reply:
`require()` returns number in react-native world. I think the fix might be like
```
let source = this.props.source;
if (this.props.source && typeof this.props.source == 'object') {
    source = Object.assign({}, this.props.source, {
        sendCookies: this.props.sendCookies,
        customUserAgent: this.props.customUserAgent || this.props.userAgent
    });

    if (this.props.html) {
        source.html = this.props.html;
    } else if (this.props.url) {
        source.uri = this.props.url;
    }
}
```